pkg/api/utils.go

```diff
@@ -40,8 +40,12 @@ func getReportFromCacheOrGenerate[T any](c cache.Cache, cacheOptions cache.Reque
 				}).Debugf("cache hit")
 				var cr T
 				if err := json.Unmarshal(res, &cr); err != nil {
+					// we got bad data back from the cache,
+					// should we log error then fall through and call generateFn?
 					return defaultVal, []error{err}
```

Comment:
I think this behavior's right, we shouldn't get bad data in the cache and if we do we need to not hide it.  I'd probably wrap the error though, `errors.WithMessagef(err, "got bad data back from the cache.  cacheKey=%+v", cacheKey)` or similar.  Otherwise we'll just get a JSON parse error without much context.